### DISCUSSION TITLE
What this error means? ERROR @wdio/sync: TypeError: hook.apply is not a function

### DISCUSSION THREAD

**Discussion by rashad-home** [webdriverio/webdriverio#7866] [Q&A]
I am migrating from wdio/sync to wdio/async, While run my code I get this error.

```
**ERROR @wdio/sync: TypeError: hook.apply is not a function**
    at execHook (C:\Users\RashadRouzdeen\Documents\NimblexAutomationFramework\node_modules\@wdio\sync\build\executeHooksWithArgs.js:36:31)
    at async TestCaseRunner.aroundTestStep (C:\Users\RashadRouzdeen\Documents\NimblexAutomationFramework\node_modules\@wdio\cucumber-framework\node_modules\cucumber\lib\runtime\test_case_runner.js:185:28)
```

**Comment by christian-bromann** [comment]
@rashad-home hard to say where this error comes from without a minimal reproducible example.

**Reply by rashad-home** [reply]
@christian-bromann I can reproduce this error while getting this issue https://github.com/webdriverio/webdriverio/discussions/6702

**Reply by rashad-home** [reply]
Do we have any replacement for "\node_modules\@wdio\sync\build\executeHooksWithArgs.js"?

**Reply by christian-bromann** [reply]
#6702 is not an issue it describes our plans moving forward with async commands in WebdriverIO. Check out https://webdriver.io/docs/async-migration for documentation how to transition over.

**Reply by rashad-home** [reply]
Sorry my bad, I gave you a wrong link.
please refer this: link, https://github.com/webdriverio/webdriverio/discussions/6702#discussioncomment-1898268

**Reply by christian-bromann** [reply]
referring to my original comment:

> hard to say where this error comes from without a minimal reproducible example.

**Comment by zbyte64** [comment]
Having a similar issue, but it doesn't seem to produce a viable stack trace. Stack overflow seems to indicate mismatched libraries: https://stackoverflow.com/questions/65939744/wdio-sync-typeerror-hook-apply-is-not-a-function
